src/vm/peimage.cpp

```diff
@@ -909,73 +909,149 @@ PTR_PEImageLayout PEImage::GetLayoutInternal(DWORD imageLayoutMask,DWORD flags)
     {
         _ASSERTE(HasID());
 
-        if (imageLayoutMask&PEImageLayout::LAYOUT_MAPPED)
+        BOOL bIsMappedLayoutSuitable;
+        BOOL bIsFlatLayoutSuitable;
+
+#ifdef PLATFORM_UNIX
+        bIsMappedLayoutSuitable = ((imageLayoutMask & PEImageLayout::LAYOUT_MAPPED) != 0);
```

Comment:
Also, "Requested" is better than "Suitable" since the incoming flag value implies the request.